widgets/valitext/org.eclipse.nebula.widgets.valitext/src/org/eclipse/nebula/widgets/TextFieldWithEvaluation/TextFieldWithEvaluation.java

```diff
@@ -0,0 +1,264 @@
+	 * @param style
+	 *            The style of widget to construct
+	 * @param maxLength
+	 *            The maximum allowed length of the text
+	 */
+	public TextFieldWithEvaluation(Composite parent, int style, int maxLength) {
+		this(parent, style, new MaxLengthEvaluator(maxLength));
+	}
+
+	/**
+	 * Constructor which can uses the given {@link IEvaluator} to evaluate the text.
+	 * 
+	 * @param parent
+	 *            The parent composite
+	 * @param evaluator
+	 *            The {@link IEvaluator} object
+	 */
+	public TextFieldWithEvaluation(Composite parent, IEvaluator evaluator) {
+		this(parent, SWT.NONE, evaluator);
+	}
+
+	/**
+	 * Constructor which can uses the given {@link IEvaluator} to evaluate the text.
+	 * 
+	 * @param parent
+	 *            The parent composite
+	 * @param style
+	 *            The style of widget to construct
+	 * @param evaluator
+	 *            The {@link IEvaluator} object
+	 */
+	public TextFieldWithEvaluation(Composite parent, int style, IEvaluator evaluator) {
+		super(parent, style);
+
+		this.evaluator = evaluator;
+
+		create();
+		updateLayout();
+	}
+
+	/**
+	 * Returns the text of the input field.
+	 * 
+	 * @return The text
+	 */
+	public String getText() {
+		return input.getText();
+	}
+
+	/**
+	 * Sets the text of the input field. Afterwards the validation is triggered.
+	 * 
+	 * @param text
+	 *            The text to set
+	 */
+	public void setText(String text) {
+		input.setText(text);
+		validate();
+	}
+
+	/**
+	 * Adds a {@link ModifyListener} to the input field.
+	 * 
+	 * @param listener
+	 *            The listener which should be notified
+	 */
+	public void addModifyListener(ModifyListener listener) {
+		input.addModifyListener(listener);
+	}
+
+	/**
+	 * Removes the given {@link ModifyListener} from the input field.
+	 * 
+	 * @param listener
+	 *            The listener which should no longer be notified
+	 */
+	public void removeModifyListener(ModifyListener listener) {
+		input.removeModifyListener(listener);
+	}
+
+	/**
+	 * Returns all registered {@link ModifyListener} of the input field.
+	 * 
+	 * @return All registeres {@link ModifyListener} of the input field
+	 */
+	public Listener[] getModifyListeners() {
+		return  this.input.getListeners(SWT.Modify);
+	}
+
+	/**
+	 * Returns the {@link IEvaluator} of this widget.
+	 * 
+	 * @return The {@link IEvaluator} of this widget
+	 */
+	public IEvaluator getEvaluator() {
+		return evaluator;
+	}
+
+	/**
+	 * Sets a new {@link IEvaluator} for this widget
+	 * 
+	 * @param evaluator
+	 *            The {@link IEvaluator} to set
+	 */
+	public void setEvaluator(IEvaluator evaluator) {
+		checkWidget();
+
+		this.evaluator = evaluator;
+
+		if (this.hint != null) {
+			this.hint.dispose();
+		}
+		if (this.input != null) {
+			this.input.dispose();
+		}
+
+		create();
+		updateLayout();
+	}
+
+	/**
+	 * Selects all text in the input field.
+	 */
+	public void selectAll() {
+		this.input.selectAll();
```

Comment:
style nit: use this.input or input this.input is more clear because there is no name like fInput that tells maintainers it is a field.